kube-ovn-dpdk-base build not successful (dpdk 22.11.1 + ovs3.1 )

# Expected Behavior
kube-ovn dpdk feature is not support kube-ovn-1.12，because the kube-ovn-dpdk image was build last year and ovs and ovn version is not correct.
# Actual Behavior
I look the kube-ovn-base Dockerfile,have a try as below,but failed:
```bash
# syntax = docker/dockerfile:experimental
FROM ubuntu:22.04 as ovs-builder

ARG ARCH
ARG DEBIAN_FRONTEND=noninteractive
ENV SRC_DIR='/usr/src'

RUN apt update && apt install -y git curl

RUN cd /usr/src/ && \
    git clone -b branch-3.1 --depth=1 https://github.com/openvswitch/ovs.git && \
    cd ovs && \
    # fix memory leak by ofport_usage and trim memory periodically
    curl -s https://github.com/kubeovn/ovs/commit/25d71867370c9a44c66b973556338de7a4d9bad7.patch | git apply && \
    # increase election timer
    curl -s https://github.com/kubeovn/ovs/commit/31f736fb54cf00e893a23e396958883f54f4080f.patch | git apply && \
    # add fdb update logging
    curl -s https://github.com/kubeovn/ovs/commit/119ab5c7e104d25641cdf4506a359c5729acdd9a.patch | git apply && \
    # fdb: fix mac learning in environments with hairpin enabled
    curl -s https://github.com/kubeovn/ovs/commit/40d5597a9a3a09015dda2202f6aa81791c5c03f3.patch | git apply && \
    # ovsdb-tool: add optional server id parameter for "join-cluster" command
    curl -s https://github.com/kubeovn/ovs/commit/ebf61515da71fa2e23125a92859fbdb96dcbffe7.patch | git apply && \
    # Add jitter parameter patch for netem qos
    curl -s https://github.com/kubeovn/ovs/commit/2eaaf89fbf3ee2172719ed10d045fd79900edc8e.patch | git apply && \
    # fix memory leak in qos
    curl -s https://github.com/kubeovn/ovs/commit/6a4dd2f4b9311a227cc26fef7c398ae9b241311b.patch | git apply && \
    # ovsdb-tool: add command fix-cluster
    curl -s https://github.com/kubeovn/ovs/commit/f52c239f5ded40b503e4d217f916b46ca413da4c.patch | git apply

RUN cd /usr/src/ && git clone -b branch-22.12 --depth=1 https://github.com/ovn-org/ovn.git && \
    cd ovn && \
    # python: Rename build related code to ovs_build_helpers.
    curl -s https://github.com/kubeovn/ovn/commit/9d961aec6fd7ef3e3002bc34e285833279e989c2.patch | git apply && \
    # change hash type from dp_hash to hash with field src_ip
    curl -s https://github.com/kubeovn/ovn/commit/4ad8763f707ff4088ae61396c7931e8735f71f22.patch | git apply && \
    # set ether dst addr for dnat on logical switch
    curl -s https://github.com/kubeovn/ovn/commit/44875725ad6ce3cb38e4d471d540fe69ed204bff.patch | git apply && \
    # modify src route priority
    curl -s https://github.com/kubeovn/ovn/commit/da1388ece89b27012d081c31310fd577b036b071.patch | git apply && \
    # fix reaching resubmit limit in underlay
    curl -s https://github.com/kubeovn/ovn/commit/6934f1a1eb5986a904eefb560c0d6d57811453d9.patch | git apply && \
    # ovn-controller: do not send GARP on localnet for Kube-OVN ports
    curl -s https://github.com/kubeovn/ovn/commit/8af8751cdb55f582c675db921f2526b06fd3d8c0.patch | git apply && \
    # ovn-ic blacklist function not work on ipv6
    curl -s https://github.com/kubeovn/ovn/commit/78ab91005854532e7eb5c4fe6b2923ce292e3681.patch | git apply

RUN apt install -y build-essential fakeroot \
    autoconf automake bzip2 debhelper-compat dh-exec dh-python dh-sequence-python3 dh-sequence-sphinxdoc \
    graphviz iproute2 libcap-ng-dev libdbus-1-dev libnuma-dev libpcap-dev libssl-dev libtool libunbound-dev \
    openssl pkg-config procps python3-all-dev python3-setuptools python3-sortedcontainers python3-sphinx \
    libjemalloc-dev python3-pip libibverbs1 libibverbs-dev ibverbs-providers

RUN pip3 install meson ninja pyelftools

RUN cd /usr/src/ && \
    curl -o dpdk-22.11.1.tar.xz https://fast.dpdk.org/rel/dpdk-22.11.1.tar.xz && \
    tar xf dpdk-22.11.1.tar.xz && \
    export DPDK_DIR=/usr/src/dpdk-stable-22.11.1 && \
    export DPDK_BUILD=$DPDK_DIR/build && \
    cd $DPDK_DIR && \
    meson build && \
    ninja -C build && \
    ninja -C build install && \
    ldconfig

RUN cd /usr/src/ovs && \
    ./boot.sh && \
    ./configure --with-dpdk=yes && \
    rm -rf .git && \
    export DPDK_DIR=/usr/src/dpdk-stable-22.11.1 && \
    CONFIGURE_OPTS='LIBS=-ljemalloc' && \
    if [ "$ARCH" = "amd64" ]; then CONFIGURE_OPTS='CFLAGS="-O2 -g -msse4.2 -mpopcnt -fPIC"'; fi && \
    DATAPATH_CONFIGURE_OPTS='--prefix=/usr' EXTRA_CONFIGURE_OPTS=$CONFIGURE_OPTS DEB_BUILD_OPTIONS='parallel=8 nocheck' make debian-deb

RUN dpkg -i /usr/src/python3-openvswitch*.deb /usr/src/libopenvswitch*.deb

RUN cd /usr/src/ovn && \
    sed -i 's/OVN/ovn/g' debian/changelog && \
    rm -rf .git && \
    ./boot.sh && \
    CONFIGURE_OPTS='--with-ovs-build=/usr/src/ovs/_debian' && \
    if [ "$ARCH" = "amd64" ]; then CONFIGURE_OPTS="$CONFIGURE_OPTS CFLAGS='-O2 -g -msse4.2 -mpopcnt -fPIC'"; fi && \
    OVSDIR=/usr/src/ovs EXTRA_CONFIGURE_OPTS=$CONFIGURE_OPTS DEB_BUILD_OPTIONS='parallel=8 nocheck' fakeroot debian/rules binary

RUN mkdir /packages/ && \
    cp /usr/src/ovs/vswitchd/ovs-vswitchd /packages && \
    cp /usr/src/libopenvswitch*.deb /packages && \
    cp /usr/src/openvswitch-*deb /packages && \
    cp /usr/src/python3-openvswitch*deb /packages && \
    cp /usr/src/ovn-*deb /packages && \
    cd /packages && rm -f *source* *doc* *datapath* *docker* *vtep* *test* *dev*

FROM ubuntu:22.04

ARG DEBIAN_FRONTEND=noninteractive
RUN apt update && apt upgrade -y && apt install ca-certificates python3 hostname libunwind8 netbase \
        ethtool iproute2 ncat libunbound-dev procps libatomic1 kmod iptables python3-netifaces python3-sortedcontainers \
        tcpdump ipset curl uuid-runtime openssl inetutils-ping arping ndisc6 \
        logrotate dnsutils net-tools strongswan strongswan-pki libcharon-extra-plugins \
        libcharon-extauth-plugins libstrongswan-extra-plugins libstrongswan-standard-plugins -y --no-install-recommends && \
        rm -rf /var/lib/apt/lists/* && \
        rm -rf /etc/localtime

RUN mkdir -p /var/run/openvswitch && \
    mkdir -p /var/run/ovn && \
    mkdir -p /etc/cni/net.d && \
    mkdir -p /opt/cni/bin

ARG ARCH
ENV CNI_VERSION=v1.3.0
RUN curl -sSf -L --retry 5 https://github.com/containernetworking/plugins/releases/download/${CNI_VERSION}/cni-plugins-linux-${ARCH}-${CNI_VERSION}.tgz | tar -xz -C . ./loopback ./portmap ./macvlan

RUN --mount=type=bind,target=/packages,from=ovs-builder,source=/packages  \
    dpkg -i /packages/libopenvswitch*.deb && \
    dpkg -i /packages/openvswitch-*.deb && \
    dpkg -i /packages/python3-openvswitch*.deb &&\
    dpkg -i --ignore-depends=openvswitch-switch,openvswitch-common /packages/ovn-*.deb && \
    cp -f /packages/ovs-vswitchd /usr/sbin/
```
# Steps to Reproduce the Problem

1. you can use above dockerfile to have a try

# Additional Info

- Kubernetes version:

  **Output of `kubectl version`:**

  ```
  (paste your output here)
  ```

- kube-ovn version:
  
  ```
  1.12
  ```

- operation-system/kernel version:
  
  **Output of `awk -F '=' '/PRETTY_NAME/ { print $2 }' /etc/os-release`:**
  **Output of `uname -r`:**
  ```
  (paste your output here)
  ```

<!-- Any other additional information -->

build exit as below error
```bash
144.0 make[2]: Entering directory '/usr/src/ovs'
144.1 py3versions: no X-Python3-Version in control file, using supported versions
144.2 dh_strip --dbgsym-migration='openvswitch-dbg (<< 2.17~)'
144.4 f924c5af5e530f820af69d82cd7f816a045b008e
144.4 a7f0df8370f0df4c30e174c6007b1b43be8f905d
144.5 e45ccf983803cddaa627415bfca44e288b036e6d
144.5 829950c9650a6088293b68a581c692e2bd2d2382
144.5 b530a48e1a1389901da64843eb54f21ddaedf2c4
144.7 abd55dabdebd04864e48d43e1b584299d7d35892
145.2 107c5d4e37b78d758e98ff35c079f35bce810d44
145.3 c124ff73cf9c956b12a4e87efe0c9ebdfabca10c
145.4 4d6058d3c59f093d7ec9b090af1968e0cfb420dd
145.7 d726478ad990c7d3c9bcfaa622e645b744ce6225
145.9 860b1ae5a1722ace40dd623e96b100aaf528fadf
146.2 d69806a385fe666026fada4a62429f010d482ae7
147.2 make[2]: Leaving directory '/usr/src/ovs'
147.2    dh_makeshlibs
147.3    dh_shlibdeps
148.1 dpkg-shlibdeps: error: no dependency information found for /usr/local/lib/x86_64-linux-gnu/librte_vhost.so.23 (used by debian/openvswitch-switch-dpdk/usr/lib/openvswitch-switch-dpdk/ovs-vswitchd-dpdk)
148.1 Hint: check if the library actually comes from a package.
148.2 dh_shlibdeps: error: dpkg-shlibdeps -Tdebian/openvswitch-switch-dpdk.substvars debian/openvswitch-switch-dpdk/usr/lib/openvswitch-switch-dpdk/ovs-vswitchd-dpdk returned exit code 25
149.2 dh_shlibdeps: error: Aborting due to earlier error
149.2 make[1]: *** [debian/rules:8: binary] Error 25
149.2 make[1]: Leaving directory '/usr/src/ovs'
149.2 make: *** [Makefile:7195: debian-deb] Error 2
------
Dockerfile:66
--------------------
  65 |     
  66 | >>> RUN cd /usr/src/ovs && \
  67 | >>>     ./boot.sh && \
  68 | >>>     ./configure --with-dpdk=yes && \
  69 | >>>     rm -rf .git && \
  70 | >>>     export DPDK_DIR=/usr/src/dpdk-stable-22.11.1 && \
  71 | >>>     CONFIGURE_OPTS='LIBS=-ljemalloc' && \
  72 | >>>     if [ "$ARCH" = "amd64" ]; then CONFIGURE_OPTS='CFLAGS="-O2 -g -msse4.2 -mpopcnt -fPIC"'; fi && \
  73 | >>>     DATAPATH_CONFIGURE_OPTS='--prefix=/usr' EXTRA_CONFIGURE_OPTS=$CONFIGURE_OPTS DEB_BUILD_OPTIONS='parallel=8 nocheck' make debian-deb
  74 |     
--------------------
ERROR: failed to solve: process "/bin/sh -c cd /usr/src/ovs &&     ./boot.sh &&     ./configure --with-dpdk=yes &&     rm -rf .git &&     export DPDK_DIR=/usr/src/dpdk-stable-22.11.1 &&     CONFIGURE_OPTS='LIBS=-ljemalloc' &&     if [ \"$ARCH\" = \"amd64\" ]; then CONFIGURE_OPTS='CFLAGS=\"-O2 -g -msse4.2 -mpopcnt -fPIC\"'; fi &&     DATAPATH_CONFIGURE_OPTS='--prefix=/usr' EXTRA_CONFIGURE_OPTS=$CONFIGURE_OPTS DEB_BUILD_OPTIONS='parallel=8 nocheck' make debian-deb" did not complete successfully: exit code: 2
```

--- a/.github/workflows/build-kube-ovn-base-dpdk.yaml
+++ b/.github/workflows/build-kube-ovn-base-dpdk.yaml
@@ -1,8 +1,17 @@
 name: Build Base DPDK
-on: workflow_dispatch
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: "20 19 * * *"
 
 jobs:
   build-amd64:
+    strategy:
+      fail-fast: false
+      matrix:
+        branch:
+          - master
+          - release-1.12
     name: Build AMD64
     runs-on: ubuntu-22.04
     steps:
@@ -21,6 +30,12 @@ jobs:
           path: image-amd64-dpdk.tar
 
   push:
+    strategy:
+      fail-fast: false
+      matrix:
+        branch:
+          - master
+          - release-1.12
     needs:
       - build-amd64
     name: push

--- a/.github/workflows/build-x86-image.yaml
+++ b/.github/workflows/build-x86-image.yaml
@@ -119,6 +119,16 @@ jobs:
         if: needs.build-kube-ovn-base.outputs.build-base == 1
         run: docker load --input image-amd64.tar
 
+      - name: Download base-dpdk images
+        if: needs.build-kube-ovn-base-dpdk.outputs.build-base-dpdk == 1
+        uses: actions/download-artifact@v3
+        with:
+          name: kube-ovn-base-dpdk
+
+      - name: Load base-dpdk images
+        if: needs.build-kube-ovn-base-dpdk.outputs.build-base-dpdk == 1
+        run: docker load --input image-amd64-dpdk.tar
+
       - name: Build
         run: |
           go mod tidy
@@ -128,6 +138,7 @@ jobs:
             TAG=$(cat VERSION)
             docker tag kubeovn/kube-ovn-base:$TAG-amd64 kubeovn/kube-ovn-base:$TAG
             docker tag kubeovn/kube-ovn-base:$TAG-debug-amd64 kubeovn/kube-ovn-base:$TAG-debug
+            docker tag kubeovn/kube-ovn-base:$TAG-amd64-dpdk kubeovn/kube-ovn-base:$TAG-dpdk
             make build-kube-ovn
           else
             make image-kube-ovn
@@ -1971,6 +1982,7 @@ jobs:
           docker tag kubeovn/kube-ovn:$TAG kubeovn/kube-ovn-dev:$COMMIT-x86
           docker tag kubeovn/kube-ovn:$TAG kubeovn/kube-ovn:$TAG-x86
           docker tag kubeovn/kube-ovn:$TAG-debug kubeovn/kube-ovn:$TAG-debug-x86
+          docker tag kubeovn/kube-ovn:$TAG-dpdk kubeovn/kube-ovn:$TAG-dpdk-x86
           docker tag kubeovn/vpc-nat-gateway:$TAG kubeovn/vpc-nat-gateway-dev:$COMMIT-x86
           docker tag kubeovn/vpc-nat-gateway:$TAG kubeovn/vpc-nat-gateway:$TAG-x86
           docker tag kubeovn/centos7-compile:$TAG kubeovn/centos7-compile-dev:$TAG-x86
@@ -1981,6 +1993,7 @@ jobs:
           docker push kubeovn/kube-ovn:$TAG-x86
           docker push kubeovn/kube-ovn-dev:$COMMIT-x86
           docker push kubeovn/kube-ovn:$TAG-debug-x86
+          docker push kubeovn/kube-ovn:$TAG-dpdk-x86
           docker push kubeovn/vpc-nat-gateway:$TAG-x86
           docker push kubeovn/vpc-nat-gateway-dev:$COMMIT-x86
           docker push kubeovn/centos7-compile:$TAG-x86

--- a/Makefile
+++ b/Makefile
@@ -128,7 +128,7 @@ base-arm64:
 image-kube-ovn: build-go
 	docker buildx build --platform linux/amd64 -t $(REGISTRY)/kube-ovn:$(RELEASE_TAG) --build-arg VERSION=$(RELEASE_TAG) -o type=docker -f dist/images/Dockerfile dist/images/
 	docker buildx build --platform linux/amd64 -t $(REGISTRY)/kube-ovn:$(DEBUG_TAG) --build-arg BASE_TAG=$(DEBUG_TAG) -o type=docker -f dist/images/Dockerfile dist/images/
-	docker buildx build --platform linux/amd64 -t $(REGISTRY)/kube-ovn:$(RELEASE_TAG)-dpdk -o type=docker -f dist/images/Dockerfile.dpdk dist/images/
+	docker buildx build --platform linux/amd64 -t $(REGISTRY)/kube-ovn:$(RELEASE_TAG)-dpdk --build-arg VERSION=$(RELEASE_TAG) -o type=docker -f dist/images/Dockerfile.dpdk dist/images/
 
 .PHONY: image-vpc-nat-gateway
 image-vpc-nat-gateway:
@@ -162,7 +162,7 @@ push-release: release
 
 .PHONY: tar-kube-ovn
 tar-kube-ovn:
-	docker save $(REGISTRY)/kube-ovn:$(RELEASE_TAG) $(REGISTRY)/kube-ovn:$(DEBUG_TAG) -o kube-ovn.tar
+	docker save $(REGISTRY)/kube-ovn:$(RELEASE_TAG) $(REGISTRY)/kube-ovn:$(DEBUG_TAG) $(REGISTRY)/kube-ovn:$(RELEASE_TAG)-dpdk -o kube-ovn.tar
 
 .PHONY: tar-vpc-nat-gateway
 tar-vpc-nat-gateway:

